runtime/common/src/lib.rs

```diff
@@ -419,3 +419,162 @@ pub mod xcm_transactor {
 		}
 	}
 }
+
+pub mod origin {
+	use std::marker::PhantomData;
+
+	use cfg_primitives::AccountId;
+	use frame_support::{
+		dispatch::RawOrigin,
+		traits::{EitherOfDiverse, EnsureOrigin},
+	};
+	use frame_system::EnsureRoot;
+	use sp_core::Get;
+
+	type EnsureAccountOrRootOr<Account, O> =
+		EitherOfDiverse<EitherOfDiverse<EnsureAccount<Account>, EnsureRoot<AccountId>>, O>;
+
+	pub struct EnsureAccount<Account>(PhantomData<Account>);
```

Comment:
@mustermeiszer Can't we just use `EnsureSignedBy` instead of building this custom `EnsureAccount`?

Reply:
Ah nicel Changed.